ta/tpm_log_test/ta_entry.c

```diff
@@ -0,0 +1,161 @@
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+/* Called each time an instance is destroyed */
+void TA_DestroyEntryPoint(void)
+{
+}
+
+/* Called each time a session is opened */
+TEE_Result TA_OpenSessionEntryPoint(uint32_t nParamTypes __unused,
+				    TEE_Param pParams[4] __unused,
+				    void **ppSessionContext __unused)
+{
+	return TEE_SUCCESS;
+}
+
+/* Called each time a session is closed */
+void TA_CloseSessionEntryPoint(void *pSessionContext __unused)
+{
+}
+
+/* Called when a command is invoked */
+TEE_Result TA_InvokeCommandEntryPoint(void *pSessionContext __unused,
+				      uint32_t nCommandID,
+				      uint32_t nParamTypes __unused,
+				      TEE_Param pParams[4] __unused)
+{
+	unsigned char buf[LOG_SIZE];
+	const uint32_t param_types = TEE_PARAM_TYPES(
+						TEE_PARAM_TYPE_MEMREF_OUTPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Param params[TEE_NUM_PARAMS] = { 0 };
+
+	switch (nCommandID) {
+	case TA_TPM_TEST_GET_LOG:
+	/*
+	 * Test case: Right parameters.
+	 */
+		params[0].memref.buffer = (void *)buf;
+		params[0].memref.size = LOG_SIZE;
+
+		if (invoke_system_pta(PTA_SYSTEM_GET_TPM_EVENT_LOG,
+				      param_types, params) == TEE_SUCCESS) {
+			uint32_t field = 0;
+			unsigned char *digestptr = NULL;
+			unsigned int i;
+
+			MSG("Received event log with %i bytes",
+			    params[0].memref.size);
+			MSG("Parsing the event log header");
+			memcpy(&field, &buf[0], sizeof(uint32_t));
+
+			MSG("--> PCRIndex = 0x%" PRIx32, field);
+			/*
+			 * PCR Index must be 0 on the header.
+			 * Ref. Section 5.3 of TCG EFI Protocol Specification.
+			 * Family 2.0 Level 00 Revision 00.13. March 30, 2016
+			 */
+			if (field != 0) {
+				EMSG("PCRIndex must be 0");
+				return TEE_ERROR_GENERIC;
+			}
+
+			memcpy(&field, &buf[4], sizeof(uint32_t));
+			MSG("--> EventType = 0x%" PRIx32, field);
+			/*
+			 * Event type must be EV_NO_ACTION on the header.
+			 * Ref. Section 5.3 of TCG EFI Protocol Specification.
+			 * Family 2.0 Level 00 Revision 00.13. March 30, 2016
+			 */
+			if (field != 3) {
+				EMSG("EventType must be 3 (EV_NO_ACTION)");
+				return TEE_ERROR_GENERIC;
+			}
+
+			/*
+			 * Header digest must be formed of 20 bytes set to 0.
+			 * Ref. Section 5.3 of TCG EFI Protocol Specification.
+			 * Family 2.0 Level 00 Revision 00.12. March 30, 2016
+			 */
+			digestptr = &buf[8];
+			for (i = 0; i < 20; i++) {
+				if (*(digestptr + i) != 0) {
+					EMSG("Digest byte %i must be 0", i);
+					return TEE_ERROR_GENERIC;
+				}
+			}
+			MSG("--> Digest value passed");
+			return TEE_SUCCESS;
+		}
+		return TEE_ERROR_ACCESS_DENIED;
+
+	case TA_TPM_TEST_SHORT_BUF:
+	/*
+	 * Test case: Insufficient space on the buffer.
+	 */
+		params[0].memref.buffer = (void *)buf;
+		params[0].memref.size = 1;
+		memset(buf, 0, LOG_SIZE);
+
+		if (invoke_system_pta(PTA_SYSTEM_GET_TPM_EVENT_LOG,
+				      param_types, params) == TEE_SUCCESS) {
```

Comment:
Why not expect `== TEE_ERROR_SHORT_BUFFER`? Other error codes would indicate a problem so you don't want to report success in this case.

Reply:
Comments fixed. I also renamed the test to 1024, as test 1023 was in use.

Thanks